### PR TITLE
Implemented Left shift for lfortran

### LINKED ISSUE #3137: Implement `lshift`
```fortran
program main
   print*, lshift(4, 2)
end
```
```console
$ gfortran a.f90 && ./a.out
          16
$ lfortran a.f90
semantic error: Function 'lshift' not found or not implemented yet (if it is intrinsic)
 --> a.f90:2:12
  |
2 |    print*, lshift(4, 2)
  |            ^^^^^^^^^^^^ 


Note: Please report unclear or confusing messages as bugs at
https://github.com/lfortran/lfortran/issues.
```
Refer: https://gcc.gnu.org/onlinedocs/gcc-4.4.7/gfortran/LSHIFT.html'
Follow: https://github.com/lfortran/lfortran/pull/3134

### PR DESCRIPTION
**Pull request by parth121101**
Fix #3137

### CODE REVIEW

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Thanks for the PR @parth121101, you may remove `shiftl` implementation from `src/runtime/impure/lfortran_intrinsic_bit.f90`, build it and push changes after performing.

```
rm -rf ./tests/reference/*
./run_tests.py -u
```

Also, follow `integration_tests/intrinsics_108.f90` and add a new test for `shiftl`.

**Comment by Pranavchiku** [comment]
Looks good! Can you apply the following diff and add new integration test calling `lshift` as well? ( Just replace `shiftl` with `lshift`, in a new test file and register )

```diff
diff --git a/src/lfortran/semantics/ast_common_visitor.h b/src/lfortran/semantics/ast_common_visitor.h
index 978b9c0d0..0026e4361 100644
--- a/src/lfortran/semantics/ast_common_visitor.h
+++ b/src/lfortran/semantics/ast_common_visitor.h
@@ -714,6 +714,7 @@ public:
         {"hypot", {IntrinsicSignature({"x", "y"}, 2, 2)}},
         {"shiftr", {IntrinsicSignature({"i", "shift"}, 2, 2)}},
         {"shiftl", {IntrinsicSignature({"i", "shift"}, 2, 2)}},
+        {"lshift", {IntrinsicSignature({"i", "shift"}, 2, 2)}},
     };
 
     std::map<std::string, std::string> double_precision_intrinsics = {
diff --git a/src/libasr/pass/intrinsic_function_registry.h b/src/libasr/pass/intrinsic_function_registry.h
index 803b4c50f..bcb3f01ab 100644
--- a/src/libasr/pass/intrinsic_function_registry.h
+++ b/src/libasr/pass/intrinsic_function_registry.h
@@ -4668,6 +4668,7 @@ namespace IntrinsicScalarFunctionRegistry {
                 {"trailz", {&Trailz::create_Trailz, &Trailz::eval_Trailz}},
                 {"shiftr", {&Shiftr::create_Shiftr, &Shiftr::eval_Shiftr}},
                 {"shiftl", {&Shiftl::create_Shiftl, &Shiftl::eval_Shiftl}},
+                {"lshift", {&Shiftl::create_Shiftl, &Shiftl::eval_Shiftl}},
                 {"leadz", {&Leadz::create_Leadz, &Leadz::eval_Leadz}},
                 {"hypot", {&Hypot::create_Hypot, &Hypot::eval_Hypot}},
                 {"kind", {&Kind::create_Kind, &Kind::eval_Kind}},
```

**Comment by HarshitaKalani** [comment]
I guess `lshift` and `shiftl` are same. But there's a problem with the suggestion I believe.
In case of any error, it'll print "this is the error in case of `shiftl`" even when the user has used `lshift`.

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -1755,6 +1756,77 @@ namespace Shiftr {
...
+
+    static inline ASR::expr_t* instantiate_Shiftl(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_sign_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("x", arg_types[0]);
+        fill_func_arg("y", arg_types[1]);
+        auto result = declare(fn_name, return_type, ReturnVar);
+        /*
+        * r = shiftr(x, y)
```
Comment (Pranavchiku) [inline src/libasr/pass/intrinsic_function_registry.h:1815]:
```suggestion
        * r = shiftl(x, y)
```

**Review by Pranavchiku — APPROVED** [review]
left a minor comment, rest all looks good to me. Thanks!

File: src/libasr/pass/intrinsic_function_registry.h [hunk 2]
```diff
@@ -1755,6 +1756,77 @@ namespace Shiftr {
...
+        }
+        return ASR::make_IntrinsicScalarFunction_t(al, loc,
+            static_cast<int64_t>(IntrinsicScalarFunctions::Shiftl),
+            args.p, args.n, 0, ASRUtils::expr_type(args[0]), m_value);
+    }
+
+    static inline ASR::expr_t* instantiate_Shiftl(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_sign_" + type_to_str_python(arg_types[0]));
```
Comment (HarshitaKalani) [inline src/libasr/pass/intrinsic_function_registry.h:1810]:
```suggestion
        declare_basic_variables("_lcompilers_shiftl_" + type_to_str_python(arg_types[0]));
```

**Review by Pranavchiku — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**